### PR TITLE
Remove restore specific already obsolete core APIs 

### LINKED ISSUE #10940: Remove obsolete core APIs
This task is meant to do a pass to remove obsolete APIs that are arguably *not* debatable.

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10940

Regression? Last working version:

## Description

This is a breaking change. 

This is only 1 pass removing APIs, and I tried to focus on APIs that are restore specific. 

There are other APIs that *are* obsolete, but I did not remove, because I did not feel I had all the context for those. 

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by zivkan** [comment]
I had a read of all the methods that are being deleted, and I think it's fine 👍

**Comment by nkolev92** [comment]
Ready for full review @zivkan

**Comment by nkolev92** [comment]
All tests passed, but had to rebase.

File: test/NuGet.Core.Tests/NuGet.ProjectModel.Test/DependencyGraphSpecTests.cs [hunk 1]
```diff
@@ -214,18 +163,18 @@ public void Load_WithPath_WhenJsonIsInvalidDgSpec_Throws(string json)
...
+        //    using (Test test = Test.Create(json))
+        //    {
+        //        DependencyGraphSpec dgSpec = DependencyGraphSpec.Load(test.FilePath);
 
-                Assert.Equal("{}", dgSpec.Json.ToString());
-            }
-        }
+        //        Assert.Equal("{}", dgSpec.Json.ToString());
+        //    }
+        //}
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.ProjectModel.Test/DependencyGraphSpecTests.cs]: I'd delete this test, rather than comment it out.

It's a rather strange edge case. Also, I don't understand why this behaviour would be desirable, let alone prevented from changing. If the file is invalid (for example it contains more than one top-level json token), why would we even want DGSpec.Load to succeed?

I think it would be better to treat it as a missing DGSpec, and fail the no-op check, thereby doing a full restore. Anyway, this PR isn't about fixing weird tests, but I think deleting this test is fine. Leaving it in the code, but commented out is an unusual choice. Why not add a `Skipped` reason to the `Theory` attribute?

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.ProjectModel.Test/DependencyGraphSpecTests.cs]: Meant to remove it. 

That test is asserting a behavior that's not relevant anymore

**Review by zivkan — APPROVED** [review]
